### DISCUSSION TITLE
Unknown command was in my container history

### DISCUSSION THREAD

**Discussion by bamsutler** [community/community#28952] [Codespaces]
Upon logging into my codespace today I noticed that something had run an `printenv > /tmp/tmp-<randomnumber>txt` command in my container. What would have don't this and why is it logging the environment?

**Comment by legomushroom** [comment]
Hey @bamsutler 👋 I'm not aware of anything on the Codespaces side that can do that 🤔 

Few questions that can help to debug this:

- Did you see the aforementioned command in some output or you only noticed the file in the temp directory?
- Can you share the repo you've created this Codespace from?
- Did you try to look at the Codespace creation logs to see if any of the steps in the Docker image or any creation commands running that? You can get the creation logs by running the `> Codespaces: View Creation Log` in the VSCode's command palette.

**Comment by bamsutler** [comment]
Hey Oleg, thanks for your response.

It's a private repo I am working in. I noticed the command we in the bash history when I logged into the code space today but did not see the file.

I am concerned because I know I did not runt his command and it seems like a security risk to have something exporting environment variables.

Cheers

**Comment by joshaber** [comment]
Some other places it could be coming from are dependencies in your devcontainer image or from your VS Code extensions (assuming you're using VS Code!).

For example, I found [this](https://github.com/microsoft/vscode-jupyter/blob/32d07f19d0eee3a3cdd3b77561035b5f92c91cb4/src/platform/terminals/helper.node.ts#L91) in VS Code's Jupyter extension which looks suspiciously similar. Is it possible that extension is/was installed in the codespace?

**Comment by rxdelta** [comment]
The same Issue, but in my current user bash history! `printenv > "/tmp/tmp-536376lGMZEihR25iM.txt"`
Is this related to VS Code? The only strange thing happened yesterday was that Plugin "Continue - open-source AI code assistant" which I cannot remember when I installed it, requires restart. and sometimes after that, this has happened. And maybe this extension is totally irrelevant! Does any one have any idea?
